src/python-common/ceph/deployment/service_spec.py

```diff
@@ -543,6 +544,10 @@ def __init__(self,
         if config:
             self.config = {k.replace(' ', '_'): v for k, v in config.items()}
 
+        self.extra_container_args: Optional[Dict[str, str]] = None
```

Comment:
I'm not sure a dict is the correct data structure here:

* `--env` can be added multiple times as in `--env NAME1=val1 --env NAME2=val2`
* `--no-healthcheck` or `--oom-kill-disable` dont not have any value

Also have a look at `args` in https://docs.ceph.com/en/latest/cephadm/services/custom-container/